src/rosdistro/distribution_cache_generator.py

```diff
@@ -193,7 +193,11 @@ def _get_cached_distribution(index, dist_name, preclean=False, ignore_local=Fals
         # since format 2 of the index file might contain a single value rather then a list
         if not isinstance(rel_file_data, list):
             rel_file_data = [rel_file_data]
-        # update cache with current distribution file
+        # if we're not including the source portion of the cache, strip it out of the existing cache
+        # in order to skip the potentially lengthy cache invalidation process.
+        if not include_source:
+            cache.source_repo_package_xmls = {}
+        # update cache with current distribution file, which filters existing cache by validity.
```

Comment:
This is the one piece I'm not certain about. We have a small enough rosdistro internally that having to constantly ls-remote all the sources to keep the cache fresh is a fine tradeoff. However, for the buildfarm, you would probably want a `--retain-cache` option, which doesn't freshen the source manifest cache, but doesn't drop it either.

In any case, since `--include-source` is an opt-in right now, I think this is fine to merge as-is. At such time as you're wanting to enable source caching by default in ros_buildfarm, we could look at an option to allow only freshening the source cache once a day or whatever.